unlock-app/src/hooks/useAuthenticate.ts

```diff
@@ -87,10 +87,10 @@ export function useAuthenticate() {
   // Signs the user out (removes the session)
   const signOut = async () => {
     try {
-      setAccount(undefined)
       await privyLogout()
       await signOutToken()
       await Promise.all([queryClient.invalidateQueries(), refetchSession()])
+      setAccount(undefined)
```

Comment:
Arguably this means logging out will take a bit more time, but I think that is ok.

Reply:
yeah it would. definitely!